tests/ut/test_decorators.py

```diff
@@ -197,6 +198,16 @@ class Dummy:
             assert memory_mock_cache.get.call_count == 0
             assert memory_mock_cache.set.call_count == 0
 
+    @pytest.mark.asyncio
+    async def test_cached_keeps_signature(self):
+        @cached()
+        async def what(self, a, b):
+            return "1"
+
+        assert what.__name__ == "what"
+        assert str(inspect.signature(what)) == '(self, a, b)'
+        assert inspect.getfullargspec(what.__wrapped__).args == ['self', 'a', 'b']
```

Comment:
Looks good. Iirc Zope uses inspect as you have since it's the only sane way to do it. There are probably some insane ways but why worry about those :-)